src/terraingenerator/main.cpp

```diff
@@ -319,27 +460,31 @@ int main(int argc, char *argv[])
             if (!terrains.contains(terrain->name()))
                 terrains.insert(terrain->name(), terrain);
 
-    // Assign terrain priority based on the order in which we encounter
-    // them (this will determine drawing order).
-    QStringList terrainOrder;
-    terrainOrder << "LavaRock";
-    terrainOrder << "Dirt2";
-    terrainOrder << "Dirt";
-    terrainOrder << "Sand";
-    terrainOrder << "Water";
-    terrainOrder << "Hole";
-    terrainOrder << "Grass";
-    terrainOrder << "PlowedSoil";
-    terrainOrder << "Lava";
+    // Setup terrain priorities.
     TerrainLessThan lessThan;
     int priority = 0;
-    foreach (const QString &terrainName, terrainOrder) {
+    foreach (const QString &terrainName, options.terrainPriority) {
         lessThan.terrainPriority.insert(terrainName, priority);
-        ++priority;
+        priority++;
```

Comment:
Please don't arbitrarily change code around. It was fine in prefix notation since the result is not used.